HLTrigger/special/plugins/HLTPPSCalFilter.cc

```diff
@@ -0,0 +1,183 @@
+      pixelLocalTrackToken_(consumes<edm::DetSetVector<CTPPSPixelLocalTrack>>(pixelLocalTrackInputTag_)),
+
+      stripLocalTrackInputTag_(iConfig.getParameter<edm::InputTag>("stripLocalTrackInputTag")),
+      stripLocalTrackToken_(consumes<edm::DetSetVector<TotemRPLocalTrack>>(stripLocalTrackInputTag_)),
+
+      diamondLocalTrackInputTag_(iConfig.getParameter<edm::InputTag>("diamondLocalTrackInputTag")),
+      diamondLocalTrackToken_(consumes<edm::DetSetVector<CTPPSDiamondLocalTrack>>(diamondLocalTrackInputTag_)),
+
+      minTracks_(iConfig.getParameter<int>("minTracks")),
+      maxTracks_(iConfig.getParameter<int>("maxTracks")),
+
+      do_express_(iConfig.getParameter<bool>("do_express")) {}
+
+bool HLTPPSCalFilter::filter(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  // Helper tool to count valid tracks
+  const auto valid_trks = [](const auto& trk) { return trk.isValid(); };
+
+  // maps for assigning filter pass / fail
+  std::unordered_map<uint32_t, bool> pixel_filter_;
+  std::unordered_map<uint32_t, bool> strip_filter_;
+  std::unordered_map<uint32_t, bool> diam_filter_;
+
+  // pixel map definition
+  pixel_filter_[CTPPSPixelDetId(0, 0, 3)] = false;
+  pixel_filter_[CTPPSPixelDetId(0, 2, 3)] = false;
+  pixel_filter_[CTPPSPixelDetId(1, 0, 3)] = false;
+  pixel_filter_[CTPPSPixelDetId(1, 2, 3)] = false;
+
+  // strip map definition
+  // diamond map definition
+
+  // First filter on pixels (2017+) selection
+  if (!pixel_filter_.empty()) {
+    edm::Handle<edm::DetSetVector<CTPPSPixelLocalTrack>> pixelTracks;
+    iEvent.getByToken(pixelLocalTrackToken_, pixelTracks);
+
+    for (const auto& rpv : *pixelTracks) {
+      if (pixel_filter_.count(rpv.id) == 0) {
+        continue;
+      }
+      auto& fltr = pixel_filter_.at(rpv.id);
+      fltr = true;
+
+      const auto ntrks = std::count_if(rpv.begin(), rpv.end(), valid_trks);
+      if (minTracks_ > 0 && ntrks < minTracks_)
+        fltr = false;
+      if (maxTracks_ > 0 && ntrks > maxTracks_)
+        fltr = false;
+    }
+
+    // compilation of filter conditions
+    if (do_express_) {
+      return (pixel_filter_.at(CTPPSPixelDetId(0, 0, 3)) && pixel_filter_.at(CTPPSPixelDetId(0, 2, 3))) ||
+             (pixel_filter_.at(CTPPSPixelDetId(1, 0, 3)) && pixel_filter_.at(CTPPSPixelDetId(1, 2, 3)));
+    } else {
+      return (pixel_filter_.at(CTPPSPixelDetId(0, 0, 3)) || pixel_filter_.at(CTPPSPixelDetId(0, 2, 3))) ||
+             (pixel_filter_.at(CTPPSPixelDetId(1, 0, 3)) || pixel_filter_.at(CTPPSPixelDetId(1, 2, 3)));
+    }
+  }
+
+  // Then filter on strips (2016-17) selection
+  if (!strip_filter_.empty()) {
+    edm::Handle<edm::DetSetVector<TotemRPLocalTrack>> stripTracks;
+    iEvent.getByToken(stripLocalTrackToken_, stripTracks);
+
+    for (const auto& rpv : *stripTracks) {
+      if (strip_filter_.count(rpv.id) == 0)
+        continue;
+      auto& fltr = strip_filter_.at(rpv.id);
+      fltr = true;
```

Comment:
If fltr is set to true always, then why the previous line?

Reply:
I am not sure what this comment is pointing out, but as mentioned in the previous reply, the strip / diam conditions can be removed, which would remove the problem for these detectors. Does this issue also apply to the case of the pixels? If so, please elaborate on what the issue is as I don't understand it at the moment.

Reply:
You set fltr to strip_filter something, and then the line immediatly afterwards you set fltr to true, so it gets reassigned a constant value...